core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -634,9 +647,13 @@ public RetentionStrategy getRetentionStrategy() {
      */
     private void closeChannel() {
         // TODO: race condition between this and the setChannel method.
-        Channel c = channel;
-        channel = null;
-        isUnix = null;
+        Channel c;
+        synchronized (channelLock) {
```

Comment:
We need to clear the absolute path atomically with closing the channel, this just adds isUnix into that atomic operation

Sent from my iPhone

> On 16 Mar 2015, at 19:13, Jesse Glick <notifications@github.com> wrote:
> 
> In core/src/main/java/hudson/slaves/SlaveComputer.java:
> 
> > @@ -634,9 +647,13 @@ public RetentionStrategy getRetentionStrategy() {
> >       */
> >      private void closeChannel() {
> >          // TODO: race condition between this and the setChannel method.
> > -        Channel c = channel;
> > -        channel = null;
> > -        isUnix = null;
> > +        Channel c;
> > +        synchronized (channelLock) {
> How is this related to the rest of the PR?
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "engineering-code-reviews" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to engineering-code-reviews+unsubscribe@cloudbees.com.
> To post to this group, send email to engineering-code-reviews@cloudbees.com.
> To view this discussion on the web visit https://groups.google.com/a/cloudbees.com/d/msgid/engineering-code-reviews/jenkinsci/jenkins/pull/1606/r26517858%40github.com.